h2/src/main/org/h2/mvstore/tx/TransactionStore.java

```diff
@@ -62,17 +63,38 @@
 
     private final DataType dataType;
 
-    private final BitSet openTransactions = new BitSet();
+    final AtomicReference<VersionedBitSet> openTransactions = new AtomicReference<>(new VersionedBitSet());
```

Comment:
Some javadoc comment about meaning of this field and its relationship with `transactions` would be nice. Logic is quite complicated here.